pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +613,291 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
+	reqBodyAssetTable := "&trigger_actions=true&dispatch.auto_cancel=" + apiResponseMCAssetTableBuild.DispatchAutoCancel + "&dispatch.buckets=" + strconv.FormatInt(apiResponseMCAssetTableBuild.DispatchBuckets, 10) + "&dispatch.enablePreview=true"
+	endpoint := fmt.Sprintf("%s", c.ManagementURI) + "/servicesNS/nobody/splunk_monitoring_console/saved/searches/DMC%20Asset%20-%20Build%20Full/dispatch"
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(reqBodyAssetTable))
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	err = c.Do(request, "201", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//UISettings is the struct for storing monitoring console app UI settings
+type UISettings struct {
+	EaiData     string `json:"eai:data"`
+	Disabled    bool   `json:"disabled"`
+	EaiACL      string `json:"eai:acl"`
+	EaiAppName  string `json:"eai:appName"`
+	EaiUserName string `json:"eai:userName"`
+}
+
+//GetMonitoringConsoleUISettings do a Get for app UI settings
+func (c *SplunkClient) GetMonitoringConsoleUISettings() (*UISettings, error) {
+	apiResponseUISettings := struct {
+		Entry []struct {
+			Content UISettings `json:"content"`
+		} `json:"entry"`
+	}{}
+	path := "/servicesNS/nobody/splunk_monitoring_console/data/ui/nav/default.distributed"
+	err := c.Get(path, &apiResponseUISettings)
+	if err != nil {
+		return nil, err
+	}
+	if len(apiResponseUISettings.Entry) < 1 {
+		return nil, err
+	}
+	return &apiResponseUISettings.Entry[0].Content, nil
+}
+
+//UpdateLookupUISettings updates assets.csv
+func (c *SplunkClient) UpdateLookupUISettings(configuredPeers string, apiResponseUISettings *UISettings) error {
+	reqBodyMCLookups := "configuredPeers=" + configuredPeers + "&eai:appName=" + apiResponseUISettings.EaiAppName + "&eai:acl=" + apiResponseUISettings.EaiACL + "&eai:userName=" + apiResponseUISettings.EaiUserName + "&disabled=" + strconv.FormatBool(apiResponseUISettings.Disabled)
+	endpoint := fmt.Sprintf("%s/servicesNS/nobody/splunk_monitoring_console/configs/conf-splunk_monitoring_console_assets/settings", c.ManagementURI)
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(reqBodyMCLookups))
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	err = c.Do(request, "200", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//UpdateMonitoringConsoleApp updates the monitoring console app
+func (c *SplunkClient) UpdateMonitoringConsoleApp() error {
+	endpoint := fmt.Sprintf("%s/servicesNS/nobody/system/apps/local/splunk_monitoring_console", c.ManagementURI)
+	request, err := http.NewRequest("POST", endpoint, nil)
+	if err != nil {
+		return err
+	}
+	err = c.Do(request, "200", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//ClusterInfo is the struct for checking ClusterInfo
+type ClusterInfo struct {
+	MultiSite string `json:"multisite"`
+}
```

Comment:
Nit: Probably, move all the structure declarations to the beginning of the file?

Reply:
I think most of our existing code is following to have the declarations at the beginning. And we can follow the same.